### ISSUE TITLE
Provision ubuntu nodes and configure them to inter node communication.

### ISSUE THREAD

**Issue by hemenkapadia** [hemenkapadia/vagrant-hadoop-cluster#3]
Provision 4 ubuntu (puppetlabs/ubuntu-12.04-64-puppet) nodes using vargrant and then
1. Install Git, ruby and librarian-puppet - to enable puppet installation in further steps
2. Set timezone to America/Los_Angeles
3. Create a user and group 'hadoop' which is the user with which all hadoop services will run and also be the user setup for inter node communication.
4. Create ssh keys and enable password less inter node communication for hadoop user.

**Comment by hemenkapadia** [comment]
Looks like working as per the test below

``` ShellSession
hekapadi@HEKAPADI-W7-3 ~/Workspaces/Personal/vagrant-hadoop-cluster (hemen_initial)
$ vagrant ssh master.local
Welcome to Ubuntu 12.04.5 LTS (GNU/Linux 3.13.0-32-generic x86_64)

 * Documentation:  https://help.ubuntu.com/
New release '14.04.2 LTS' available.
Run 'do-release-upgrade' to upgrade to it.

Last login: Sun Oct 18 13:59:09 2015 from 10.0.2.2
vagrant@master:~$ sudo su hadoop
hadoop@master:/home/vagrant$ whoami
hadoop
hadoop@master:/home/vagrant$ ssh hadoop1.local
Warning: Permanently added 'hadoop1.local,192.168.48.11' (ECDSA) to the list of known hosts.
Welcome to Ubuntu 12.04.5 LTS (GNU/Linux 3.13.0-32-generic x86_64)

 * Documentation:  https://help.ubuntu.com/
New release '14.04.2 LTS' available.
Run 'do-release-upgrade' to upgrade to it.


The programs included with the Ubuntu system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Ubuntu comes with ABSOLUTELY NO WARRANTY, to the extent permitted by
applicable law.

hadoop@hadoop1:~$ uname -a
Linux hadoop1 3.13.0-32-generic #57~precise1-Ubuntu SMP Tue Jul 15 03:51:20 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
hadoop@hadoop1:~$ exit
logout
Connection to hadoop1.local closed.
hadoop@master:/home/vagrant$ ssh hadoop2.local
Warning: Permanently added 'hadoop2.local,192.168.48.12' (ECDSA) to the list of known hosts.
Welcome to Ubuntu 12.04.5 LTS (GNU/Linux 3.13.0-32-generic x86_64)

 * Documentation:  https://help.ubuntu.com/
New release '14.04.2 LTS' available.
Run 'do-release-upgrade' to upgrade to it.


The programs included with the Ubuntu system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Ubuntu comes with ABSOLUTELY NO WARRANTY, to the extent permitted by
applicable law.

hadoop@hadoop2:~$ uname -a
Linux hadoop2 3.13.0-32-generic #57~precise1-Ubuntu SMP Tue Jul 15 03:51:20 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
hadoop@hadoop2:~$ exit
logout
Connection to hadoop2.local closed.
hadoop@master:/home/vagrant$ ssh hadoop3.local
Warning: Permanently added 'hadoop3.local,192.168.48.13' (ECDSA) to the list of known hosts.
Welcome to Ubuntu 12.04.5 LTS (GNU/Linux 3.13.0-32-generic x86_64)

 * Documentation:  https://help.ubuntu.com/
New release '14.04.2 LTS' available.
Run 'do-release-upgrade' to upgrade to it.


The programs included with the Ubuntu system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Ubuntu comes with ABSOLUTELY NO WARRANTY, to the extent permitted by
applicable law.

hadoop@hadoop3:~$ uname -a
Linux hadoop3 3.13.0-32-generic #57~precise1-Ubuntu SMP Tue Jul 15 03:51:20 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
hadoop@hadoop3:~$ date
Sun Oct 18 14:04:32 PDT 2015
hadoop@hadoop3:~$ exit
logout
Connection to hadoop3.local closed.
hadoop@master:/home/vagrant$ date
Sun Oct 18 14:04:44 PDT 2015
hadoop@master:/home/vagrant$ exit
exit
vagrant@master:~$ exit
logout
Connection to 127.0.0.1 closed.

hekapadi@HEKAPADI-W7-3 ~/Workspaces/Personal/vagrant-hadoop-cluster (hemen_initial)
$
```

**Comment by hemenkapadia** [comment]
Closed as part of commit 519cdb02c5ca6224cbad0a615e066ec296e5a509 in hemen_inital branch
